src/eos-updater-data.c

```diff
@@ -22,17 +22,31 @@
 
 #include "eos-updater-data.h"
 
+#include "eos-util.h"
+
 #include <string.h>
 
-void
+gboolean
 eos_updater_data_init (EosUpdaterData *data,
-                       OstreeRepo *repo)
+                       OstreeRepo *repo,
+                       GError **error)
 {
-  g_return_if_fail (data == NULL);
-  g_return_if_fail (OSTREE_IS_REPO (repo));
+  if (data == NULL)
+    return TRUE;
```

Comment:
Apparently I wasn't able to come up with a single convention for initializing the structure, that is - whether it is acceptable to pass NULL here or not.

Well, I guess it should be treated as a programmer error, so I'd use `g_return_val_if_fail` as it used to be.